File: 0055-jump-game/0055-jump-game.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var canJump = function(nums) {
    let end = nums.length -1 
    for (let i = end; i >= 0;i--){
        if (i + nums[i] >= end){
            end = i
        }
    }
    return end == 0
};

File: 0055-jump-game/0055-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        end = len(nums)- 1
        dp = {}
        for i in range(len(nums) - 1, -1, -1):
            if i + nums[i] >= end:
                end = i 
        return end == 0 
                